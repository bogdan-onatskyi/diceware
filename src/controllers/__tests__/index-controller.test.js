import React from 'react';

import IndexController from '../index-controller';
import passwordObject from '../../models/password-object';

describe("IndexController component", () => {
    it("should match saved snapshot", () => {
        passwordObject.wordArray.forEach((wordObject) => {
            wordObject.index = 0;
        });
        const wrapper = shallow(
            <IndexController passwordObject={passwordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should have 1 prop", () => {
        const propsStr = 'passwordObject';
        const props = [];
        for (let prop in IndexController.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});