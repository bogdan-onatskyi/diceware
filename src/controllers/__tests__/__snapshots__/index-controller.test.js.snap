// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexController component should match saved snapshot 1`] = `
<div
  className="app__content"
>
  <div
    className="words-view"
  >
    <_class
      key="WordController_1"
      wordObject={
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 1,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        }
      }
    />
    <_class
      key="WordController_2"
      wordObject={
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 2,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        }
      }
    />
    <_class
      key="WordController_3"
      wordObject={
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 3,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        }
      }
    />
    <_class
      key="WordController_4"
      wordObject={
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 4,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        }
      }
    />
    <_class
      key="WordController_5"
      wordObject={
        Word {
          "_filter": "",
          "_index": 0,
          "_indexFiltered": Array [],
          "_wordViewId": 5,
          "editorOpened": false,
          "handleBackspace": [Function],
          "handleClick": [Function],
          "handleCountWords": [Function],
          "handleFilter": [Function],
          "handleWheel": [Function],
          "toggleEditor": [Function],
        }
      }
    />
  </div>
  <_class
    passwordObject={
      Password {
        "_maxWords": 8,
        "_usedWords": 5,
        "caps": 63,
        "handleResetAllWords": [Function],
        "handleUsedWords": [Function],
        "init": [Function],
        "isCAPS": [Function],
        "isVariantsOpened": false,
        "separatedPassword": [Function],
        "separators": Array [
          "-",
          "_",
          ":",
          ".",
          "",
          " ",
        ],
        "toggleCAPS": [Function],
        "toggleVariantsOpened": [Function],
        "wordArray": Array [
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 1,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 2,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 3,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 4,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 5,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
        ],
      }
    }
  />
  <_class
    passwordObject={
      Password {
        "_maxWords": 8,
        "_usedWords": 5,
        "caps": 63,
        "handleResetAllWords": [Function],
        "handleUsedWords": [Function],
        "init": [Function],
        "isCAPS": [Function],
        "isVariantsOpened": false,
        "separatedPassword": [Function],
        "separators": Array [
          "-",
          "_",
          ":",
          ".",
          "",
          " ",
        ],
        "toggleCAPS": [Function],
        "toggleVariantsOpened": [Function],
        "wordArray": Array [
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 1,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 2,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 3,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 4,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
          Word {
            "_filter": "",
            "_index": 0,
            "_indexFiltered": Array [],
            "_wordViewId": 5,
            "editorOpened": false,
            "handleBackspace": [Function],
            "handleClick": [Function],
            "handleCountWords": [Function],
            "handleFilter": [Function],
            "handleWheel": [Function],
            "toggleEditor": [Function],
          },
        ],
      }
    }
  />
</div>
`;
